environ_test.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadMatrixFromEnviron(t *testing.T) {
+
+	clients := []string{"yarpc-go", "yarpc-node", "yarpc-browser"}
+	servers := []string{"yarpc-go", "yarpc-node"}
+	transports := []string{"http", "tchannel"}
+
+	os.Setenv("XLANG_CLIENTS", strings.Join(clients, ","))
```

Comment:
I might switch this to `os.Setenv("XLANG_CLIENTS", "yarpc-go,yarpc-node,yarpc-browser")` to favor readability, and just duplicate the values in the assert statement.